From 0b2b06449a37682bfbb0a65c98e86cbcc4f447ef Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Wed, 4 Sep 2024 10:30:04 +0200
Subject: [PATCH 1085/1695] core: scmi-server: update SCP-Firmware version
 display

When _CFG_SCP_FIRMWARE_IN_TREE=y, the SCP Firmware sources are
inside OP-TEE source tree. When using Git, SCP-firmware prints the OP-TEE
Git revision instead of the SCP-Firmware revision. Change this behavior
by displaying information on the SCP-Firmware version present in the
makefile variable scpfw-integ-version followed by OP-TEE revision.

Before:
[    0.000000] SCP-firmware 3.19.0

After:
[    0.000000] SCP-firmware 2.12.0-intree-optee-os-3.19.0

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 8144c112ba4ed734dd44bb27c507fdd2c90e0d8a core: scmi-server: update SCP-Firmware version display]
Change-Id: Id366b8b576cb6517dc82e4a42b52f261381a7207
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/402120
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/lib/scmi-server/sub.mk | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/core/lib/scmi-server/sub.mk b/core/lib/scmi-server/sub.mk
index 9250b606c..4fdcda70a 100644
--- a/core/lib/scmi-server/sub.mk
+++ b/core/lib/scmi-server/sub.mk
@@ -59,8 +59,12 @@ cppflags-lib-y += -DBUILD_VERSION_MAJOR=$(scpfw-integ-version-maj) \
 		  -DBUILD_VERSION_MINOR=$(scpfw-integ-version-min) \
 		  -DBUILD_VERSION_PATCH=$(scpfw-integ-version-pat)
 
+ifeq ($(_CFG_SCP_FIRMWARE_IN_TREE),y)
+scpfw-impl-version := $(shell echo $(scpfw-integ-version)-intree-optee-os-$(TEE_IMPL_VERSION))
+else
 scpfw-impl-version := $(shell git -C $(scpfw-src-path) describe --tags --always --dirty=-dev 2>/dev/null || \
-                      echo Unknown_$(scpfw-integ-version))
+			echo Unknown_$(scpfw-integ-version))
+endif
 cppflags-lib-y += -DBUILD_VERSION_DESCRIBE_STRING=\"$(scpfw-impl-version)\"
 
 cppflags-lib-y += -DFWK_LOG_LEVEL=$(CFG_SCPFW_LOG_LEVEL)
-- 
2.47.2

